### ISSUE TITLE
Regression: empty string paths no longer match

### ISSUE THREAD

**Issue by heygrady** [ReactTraining/react-router#6941]
https://github.com/ReactTraining/react-router/blob/0c058adc5b56c6d9b409af5e4d8330c645fac00c/packages/react-router/modules/matchPath.js#L38

Broken by https://github.com/ReactTraining/react-router/commit/7bd1407803ba8fbec8283f12a27128b8bd4bde75 (https://github.com/ReactTraining/react-router/pull/6715)

It is possible to have an empty-string path. Prior to the breaking change this would essentially match anything.

I changed the check to be more specific:

```js
// improve the bail to allow empty-string paths
if (!path && path !== '') return null;
```

```js
// I added this test to `matchPath-test.js`
describe('with path=""', () => {
    it('returns correct url at ""', () => {
      const path = "";
      const pathname = "/";
      const match = matchPath(pathname, path);
      expect(match.url).toBe("/"); // <-- weird 🤷🏻‍♂️ 
    });

    it('returns correct url at "/somewhere/else"', () => {
      const path = "";
      const pathname = "/somewhere/else";
      const match = matchPath(pathname, path);
      expect(match.url).toBe(""); // <-- expected
    });
  });
```

I confirmed the above changes fixed the issue for my app.

I was relying on empty-string paths in my app to enable a custom route config where I needed a conceptual route that always matched. Upgrading to v5.0.1 broke my app and I couldn't figure out why.

**Comment by timdorr** [comment]
Could you submit a PR to fix this?

**Comment by mjackson** [comment]
This was released in 5.1.1 https://github.com/ReactTraining/react-router/releases/tag/v5.1.1
